### PR TITLE
Action API: List of loans

### LINKED ISSUE #12: Action API: List of loans
- Implement action `LoanList.kt`
- Check permissions for loan's info request
- Response
---
- Create Exceptions if need
dependencies: #6, #7, #8

### PR DESCRIPTION
**Pull request by BarracudaPff**
- [x] Implement action `LoanList.kt`
- [x] Check permissions for loan's info request
- [x] Response
---
- [x] Create Exceptions if need
closes #12

### CODE REVIEW

File: src/com/apmath/loans/application/v1/actions/ListLoans.kt [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+import com.apmath.loans.domain.services.LoanServiceInterface
+import com.apmath.validation.simple.NullableValidator
+import io.ktor.application.ApplicationCall
+import io.ktor.response.respond
+
+suspend fun ApplicationCall.v1ListLoans(loanService: LoanServiceInterface) {
+    val mixed = Mixed(
+        getUserId(request),
+        request.headers["service"],
+        parameters["client"]
```
Comment (malinink) [inline src/com/apmath/loans/application/v1/actions/ListLoans.kt]: is parameters allways hashSet of strings? if no, all parameters except body must be verified separately

Comment (malinink) [reply src/com/apmath/loans/application/v1/actions/ListLoans.kt]: It must looks like creditId validation in credit project

File: src/com/apmath/loans/application/v1/models/Mixed.kt [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+import com.apmath.loans.domain.models.MixedId
+import com.apmath.loans.infrastructure.fetchers.Host
+
+data class Mixed(
+    var clientIdHeader: Int?,
+    var serviceIdHeader: String?,
+    var clientId: String?
+)
+
+fun Mixed.toMixedId() = MixedId(
```
Comment (malinink) [inline src/com/apmath/loans/application/v1/models/Mixed.kt]: I think that `MixedId` is not good enough for that parameter, may be it is a good idea pass multiple parameters to service?

File: src/com/apmath/loans/domain/models/loans/Loan.kt [hunk 3]
```diff
@@ -32,6 +33,7 @@ class Loan(
     override var completed: Boolean = false
         private set
 
+    @Transient
```
Comment (malinink) [inline src/com/apmath/loans/domain/models/loans/Loan.kt]: We are using serialization? What are the purposes?

**Review by malinink — CHANGES_REQUESTED** [review]

File: src/com/apmath/loans/application/v1/models/Mixed.kt (hunk 2 above)
Comment (malinink) [reply src/com/apmath/loans/application/v1/models/Mixed.kt]: `LoansListRequest` is better if you still want to pass single parameter

Comment (BarracudaPff) [reply src/com/apmath/loans/application/v1/models/Mixed.kt]: Changed to parameters
Thought that we pass only classes

File: src/com/apmath/loans/domain/models/loans/Loan.kt (hunk 3 above)
Comment (BarracudaPff) [reply src/com/apmath/loans/domain/models/loans/Loan.kt]: In [specification](https://docs.google.com/document/d/1RiURQhXa6QD7rrq4720uTNWcksZWoBAFmj54RSLcgUg/edit#bookmark=id.cpdcdy6gb1mp), we respond Loans without payments

Comment (malinink) [reply src/com/apmath/loans/domain/models/loans/Loan.kt]: It is domain model, when you want to return Loan you should provide application loan model and map domain loan into application loan

Comment (malinink) [reply src/com/apmath/loans/domain/models/loans/Loan.kt]: Look in calculations

File: src/com/apmath/loans/application/v1/actions/ListLoans.kt [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+import io.ktor.response.respond
+
+suspend fun ApplicationCall.v1ListLoans(loanService: LoanServiceInterface) {
+    val isService = isService(request)
+    val clientIdHeader = getUserId(request)
+
+    val clientId = try {
+        getClientAttributeId(this)
+    } catch (e: NumberFormatException) {
+        respond("Client id must be between 1 and ${Int.MAX_VALUE}")
```
Comment (malinink) [inline src/com/apmath/loans/application/v1/actions/ListLoans.kt]: spec has another error format

**Review by malinink — CHANGES_REQUESTED** [review]

File: src/com/apmath/loans/domain/models/loans/Loan.kt (hunk 3 above)
Comment (BarracudaPff) [reply src/com/apmath/loans/domain/models/loans/Loan.kt]: @malinink is it ok now?

Comment (malinink) [reply src/com/apmath/loans/domain/models/loans/Loan.kt]: @BarracudaPff looks good

**Review by malinink — APPROVED** [review]

**Pull request merged**